### ISSUE TITLE
ConciseDateFormatter ignores timezone

### ISSUE THREAD

**Issue by ihincks** [matplotlib/matplotlib#15468]
### Bug report

**Bug summary**

`matplotlib.dates.ConciseDateFormatter` accepts a `tz` kwarg representing timezone that is ignored by `format_ticks`.

**Code for reproduction**

```python
import time,datetime,matplotlib
import matplotlib.pyplot as plt
import numpy as np
import matplotlib.dates as mdates
from dateutil import tz


data = [i for i in range(13)]

start = datetime.datetime.fromtimestamp(0) 
timezone = tz.gettz("EST")

timestamps_utc = [datetime.datetime(1970, 1, 1, hour=i+5,tzinfo=tz.tzutc()) for i in range(len(data))]
timestamps_tz =[ts.astimezone(timezone) for ts in timestamps_utc]

auto = mdates.AutoDateFormatter(mdates.AutoDateLocator(), tz=timezone)
concise = mdates.ConciseDateFormatter(mdates.AutoDateLocator(), tz=timezone)


fig = plt.figure(figsize=(10.0, 8))


#now plot all variations
plt.subplot(411)
plt.gca().xaxis.set_major_formatter(auto)
plt.scatter(timestamps_utc, data)
plt.gca().set_xlim([datetime.datetime(1970,1,1), datetime.datetime(1970,1,2,12)])
plt.gca().set_title("1 - data is in UTC, formatter=AutoDateFormatter")


plt.subplot(412)
plt.gca().xaxis.set_major_formatter(concise)
plt.scatter(timestamps_utc, data)
plt.gca().set_xlim([datetime.datetime(1970,1,1), datetime.datetime(1970,1,2,12)])
plt.gca().set_title("2 - data is in EST, formatter=ConciseDateFormatter")

plt.subplot(413)
plt.gca().xaxis.set_major_formatter(auto)
plt.scatter(timestamps_tz, data)
plt.gca().set_xlim([datetime.datetime(1970,1,1), datetime.datetime(1970,1,2,12)])
plt.gca().set_title("3 - data is in UTC, formatter=AutoDateFormatter")


plt.subplot(414)
plt.gca().xaxis.set_major_formatter(concise)
plt.scatter(timestamps_tz, data)
plt.gca().set_xlim([datetime.datetime(1970,1,1), datetime.datetime(1970,1,2,12)])
plt.gca().set_title("4 - data is in EST, formatter=ConciseDateFormatter")

fig.tight_layout(pad=2)
plt.show()
```

**Actual outcome**

![Screenshot from 2019-10-21 15-45-37](https://user-images.githubusercontent.com/2229105/67237755-df58a880-f419-11e9-87d2-24d8fb7a64e2.png)

**Expected outcome**

In all of the above plots, the first dot should be on Jan 1st 1970 at 00:00. This is true when `AutoDateFormatter` is used (1st and 3rd plots). This is false when `ConciseDateFormatter` is used (2nd and 4th plots). (Somewhat obscured by differing choices of x locations by `AutoDateLocator`) This is true regardless of the timestamps' timezone (as it should be---it is the plotter's job alone to choose a timezone).

I believe [this line](https://github.com/matplotlib/matplotlib/blob/6d71c85c1763c99e454ed467c3f6c54dc614e91b/lib/matplotlib/dates.py#L755) is the culprit, and should be changed to:
```
tickdatetime = [num2date(value, self._ts) for value in values]
```

**Matplotlib version**

  * Operating system: Ubuntu 19.04
  * Matplotlib version: 3.1.1
  * Matplotlib backend (`print(matplotlib.get_backend())`): Qt5Agg / module://ipykernel.pylab.backend_inline
  * Python version: 3.7.3

matplotlib installed from pip

**Comment by jklymak** [comment]
That seems reasonable.  Are you able to provide a PR?  I can fix it if not....

**Comment by ihincks** [comment]
Thanks, I started a PR.

**Comment by jklymak** [comment]
Was closed by #15478
